packages/dds/map/src/directory.ts

```diff
@@ -1074,22 +1119,26 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         }
 
         // Create the sub directory locally first.
-        const isNew = this.createSubDirectoryCore(subdirName, true);
-
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const subDir: IDirectory = this._subdirectories.get(subdirName)!;
+        const isNew = this.createSubDirectoryCore(subdirName, true, -1, this.directory.isAttached() ?
```

Comment:
You can do (this.runtime.clientId ?? "detached") with same result (without need to test this.directory.isAttached()), I think.
Though there is a difference - this will give you clientId for attached container, but detached DDS. Not sure which one is correct,

Reply:
Yes, it will work. Change it. We just need to pass the clientId which created this directory. If its not there then "detached" should ok to pass.